File: list_repos.py
"""
Gitea List Repositories under an Organization
Lists all repos in a given organization with their clone URLs.

Usage:
  python list_repos.py <org_name>
  Example: python list_repos.py 114-2-DesignProject
"""

import sys
from config import get_credentials
from gitea_api import get_session, get_org_repos


def main():
    if len(sys.argv) < 2:
        print("Usage: python list_repos.py <org_name>")
        sys.exit(1)

    org_name = sys.argv[1]
    admin_user, admin_pass = get_credentials()
    session = get_session(admin_user, admin_pass)

    repos = get_org_repos(session, org_name)

    if not repos:
        print(f"No repositories found in '{org_name}'.")
        return

    print(f"Organization: {org_name} ({len(repos)} repos)\n")
    for repo in repos:
        print(f"  {repo['name']}")
        print(f"    URL:   {repo['html_url']}")
        print(f"    Clone: {repo['clone_url']}")
        print(f"    Private: {repo['private']}")
        print()


if __name__ == "__main__":
    main()

File: scripts/list_repos.ts
/**
 * Gitea List Repos under a Course
 *
 * Usage: node list_repos.js <course_name>
 */

import { get_credentials } from './config.ts';
import { get_session, get_org_repos } from './gitea_api.ts';

async function main(): Promise<void> {
  if (process.argv.length < 3) {
    console.log('Usage: node list_repos.js <course_name>');
    process.exit(1);
  }

  const course_name = process.argv[2];
  const [admin_user, admin_pass] = await get_credentials();
  const session = get_session(admin_user, admin_pass);

  const repos = await get_org_repos(session, course_name);

  if (!repos.length) {
    console.log(`No repos found in '${course_name}'.`);
    return;
  }

  console.log(`Course: ${course_name} (${repos.length} repos)\n`);
  for (const repo of repos) {
    console.log(`  ${repo.name}`);
    console.log(`    URL:   ${repo.html_url}`);
    console.log(`    Clone: ${repo.clone_url}`);
    console.log(`    Private: ${repo.private}`);
    console.log();
  }
}

main();
